modules/renter/contractor/allowance.go

```diff
@@ -73,6 +73,7 @@ func (c *Contractor) SetAllowance(a modules.Allowance) error {
 	c.log.Println("INFO: setting allowance to", a)
 	c.mu.Lock()
 	c.allowance = a
+	c.currentPeriod = c.blockHeight
```

Comment:
This is not quite correct behavior I think. You don't want to reset the current period every time that you change the allowance. For example, if all you are doing here is adjusting the spending amount, you would expect the current period to not change.

I'm not actually certain what you should do if you are adjusting the period. If you increase the period, should you change the currentPeriod? If you decrase it, should you change the currentPeriod? Not clear to me at the moment what the best choice is there.

Reply:
Generally I think we want to follow a pattern of sticking to a rhythmic billing cycle, for example one a month, always on the same day. But if you change from a month to 6 weeks... I'm not sure what you are supposed to do.

And, does this code properly clean up the contract lines? I think that if you yank the current period around like this you can end up with contract lines that actually span multiple periods.

Reply:
My gut right now says we should only change the current period if `c.currentPeriod+(a.Period-a.RenewWindow) < c.blockHeight`, meaning that we would want to renew instantly.

Reply:
Would we also want to start the `currentPeriod` at the current blockheight if an allowance is set and there are no previous allowances?

Reply:
In most cases, that would satisfy the above equation, but I guess there are some corner cases if your `a.Period-a.RenewWindow` is greater than the block height, you could have issues.

So yeah, seems fine to check if the allowance is blank, and if so then you set the currentPeriod to the block height.